posts/models.py

```diff
@@ -155,6 +155,18 @@ def annotate_user_last_forecasts_date(self, author_id: int):
             )
         )
 
+    def annotate_has_active_forecast(self, author_id: int):
+        """
+        Annotates if user has active forecast for post
+        """
+        return self.annotate(
+            has_active_forecast=Exists(
+                Forecast.objects.filter(
+                    post_id=OuterRef("pk"), author_id=author_id, end_time__isnull=True
+                )
+            )
+        )
```

Comment:
As far as I recall, when you submit a forecast, we always set the `end_time` of the previous forecast to today's date, while your new forecast always has `end_time=None`. Could there ever be a situation where a user has only two forecasts, both with `end_time != None`? If so, why?

Reply:
Yes, withdrawal is the exact time when you'd have your most recent foreacst with a set end_time.
So this checks if there is any active forecast - meaning one with end_time of None.
If the user withdrew, then the end time of their final forecast was set to the moment of withdrawal.